prometheus/value_test.go

```diff
@@ -108,3 +110,132 @@ func TestNewConstMetricWithCreatedTimestamp(t *testing.T) {
+			}),
+			labelValues: []string{"1", "2", "3"},
+			want: []*dto.LabelPair{
+				{Name: proto.String("label-1"), Value: proto.String("1")},
+				{Name: proto.String("label-2"), Value: proto.String("2")},
+				{Name: proto.String("label-3"), Value: proto.String("3")},
+				{Name: proto.String("var-label-1"), Value: proto.String("1")},
+				{Name: proto.String("var-label-2"), Value: proto.String("2")},
+				{Name: proto.String("var-label-3"), Value: proto.String("3")},
+			},
+		},
+		{
+			name: "unsorted variable and const labels are sorted",
+			desc: NewDesc("metric-1", "", []string{"var-label-3", "var-label-2", "var-label-1"}, map[string]string{
+				"label-3": "3",
+				"label-2": "2",
+				"label-1": "1",
+			}),
+			labelValues: []string{"3", "2", "1"},
+			want: []*dto.LabelPair{
+				{Name: proto.String("label-1"), Value: proto.String("1")},
+				{Name: proto.String("label-2"), Value: proto.String("2")},
+				{Name: proto.String("label-3"), Value: proto.String("3")},
+				{Name: proto.String("var-label-1"), Value: proto.String("1")},
+				{Name: proto.String("var-label-2"), Value: proto.String("2")},
+				{Name: proto.String("var-label-3"), Value: proto.String("3")},
+			},
+		},
+	}
+	for _, tt := range tests {
+		t.Run(tt.name, func(t *testing.T) {
+			if got := MakeLabelPairs(tt.desc, tt.labelValues); !reflect.DeepEqual(got, tt.want) {
+				t.Errorf("%v != %v", got, tt.want)
+			}
+		})
+	}
+}
+
+func Benchmark_MakeLabelPairs(b *testing.B) {
+	benchFunc := func(desc *Desc, variableLabelValues []string) {
+		MakeLabelPairs(desc, variableLabelValues)
+	}
+
+	benchmarks := []struct {
+		name                string
+		bench               func(desc *Desc, variableLabelValues []string)
+		desc                *Desc
+		variableLabelValues []string
+	}{
+		{
+			name: "1 label",
+			desc: NewDesc(
+				"metric",
+				"help",
+				[]string{"var-label-1"},
+				Labels{"const-label-1": "value"}),
+			variableLabelValues: []string{"value"},
+		},
+		{
+			name: "3 labels",
+			desc: NewDesc(
+				"metric",
+				"help",
+				[]string{"var-label-1", "var-label-3", "var-label-2"},
+				Labels{"const-label-1": "value", "const-label-3": "value", "const-label-2": "value"}),
+			variableLabelValues: []string{"value", "value", "value"},
+		},
+		{
+			name: "10 labels",
+			desc: NewDesc(
+				"metric",
+				"help",
+				[]string{"var-label-5", "var-label-1", "var-label-3", "var-label-2", "var-label-10", "var-label-4", "var-label-7", "var-label-8", "var-label-9"},
+				Labels{"const-label-4": "value", "const-label-1": "value", "const-label-7": "value", "const-label-2": "value", "const-label-9": "value", "const-label-8": "value", "const-label-10": "value", "const-label-3": "value", "const-label-6": "value", "const-label-5": "value"}),
+			variableLabelValues: []string{"value", "value", "value", "value", "value", "value", "value", "value", "value", "value"},
+		},
+	}
+	for _, bm := range benchmarks {
+		b.Run(bm.name, func(b *testing.B) {
```

Comment:
```suggestion
		b.Run(fmt.Sprintf("labels=%v", len(bm.variableLabelValues)), func(b *testing.B) {
```

(or using bm.name and changing name to number, or similar)

As per https://www.bwplotka.dev/2024/go-microbenchmarks-benchstat/